### DISCUSSION TITLE
RuntimeError: This event loop is already running

### DISCUSSION THREAD

**Discussion by calvintancy** [llm-workflow-engine/llm-workflow-engine#209] [Q&A]
Hi, i am trying to use chatgpt in python, using VS Code's jupyter plugin. I can import the ChatGPT package, but when i try to load the function, i get the error below. How do i resolve this? Thank you so much. 

RuntimeError: This event loop is already running

Extra info: Using a clean x86 conda environment, on apple M1.

**Comment by thehunmonkgroup** [comment]
We have a sync and and an async class, which are you using? Can you provide a brief example of what you're doing?

**Reply by calvintancy** [reply]
Thank you so much for replying. 

I am trying to use the wrapper in python, following the toy example. I have attached a screenshot.

I have also re-installed my conda enviroment. Using a clean ARM enviroment now.

<img width="1137" alt="Screenshot 2023-03-08 at 7 55 44 AM" src="https://user-images.githubusercontent.com/62472236/223582580-081fef6e-fcd6-4655-85b6-155ac7514f3a.png">

**Comment by thehunmonkgroup** [comment]
You may have more luck with the `AsyncChatGPT` class, or it's possible you need to research how to overcome that error using `asyncio` methods.

The `ChatGPT` class uses `AsyncChatGPT` under the hood, and I believe async and threads require some special consideration, which is not something I know much about yet.

**Comment by thehunmonkgroup** [comment]
There have been some recent updates that may help with this...maybe...

Due to limitations in Python's `asyncio`, it may be problematic or impossible to execute an async function within a sync context that's already running an event loop.
